Assets/BossRoom/Scripts/Shared/Net/UnityServices/Lobbies/LobbyAsyncRequests.cs

```diff
@@ -0,0 +1,339 @@
+﻿using System;
+using System.Collections.Generic;
+using BossRoom.Scripts.Shared.Infrastructure;
+using BossRoom.Scripts.Shared.Net.UnityServices.Infrastructure;
+using Unity.Multiplayer.Samples.BossRoom;
+using Unity.Services.Authentication;
+using Unity.Services.Lobbies.Models;
+
+namespace BossRoom.Scripts.Shared.Net.UnityServices.Lobbies
+{
+    /// <summary>
+    /// An abstraction layer between the direct calls into the Lobby API and the outcomes you actually want. E.g. you can request to get a readable list of
+    /// current lobbies and not need to make the query call directly.
+    /// </summary>
+    public class LobbyAsyncRequests : IDisposable

```

Comment:
I'm curious, why does this sit on top of LobbyAPIInterface? Why not just put all the logic here at the same level as the interface? It just seems like one more onion layer that's tightly coupled anyway with LobbyAPIInterface?

Reply:
Yeah, all of it might as well be rolled into one singular Facade.